indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -19,48 +24,135 @@ import kotlinx.serialization.encoding.Encoder
+                    @OptIn(ExperimentalStdlibApi::class)
+                    if (consumed != currentNode) throw Asn1Exception(
+                        "Trailing bytes in OID Node ${
+                            currentNode.toByteArray().toHexString(HexFormat.UpperCase)
+                        }"
+                    )
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Cursed encoding of OID nodes. A sacrifice of pristine numbers requested by past gods of the netherrealm.
+     * Lazily evaluated
+     */
+    val bytes: ByteArray by if (bytes != null) lazyOf(bytes) else lazy {
+        nodes ?: throw Asn1Exception("Empty nodes are not supported")
+        nodes.toOidBytes()
     }
 
+    /**
+     * Lazily evaluated list of OID nodes (e.g. `[1, 2, 35, 4654]`)
+     */
+    val nodes by if (nodes != null) lazyOf(nodes) else lazy {
+        bytes ?: throw Asn1Exception("Empty nodes are not supported")
+        val (first, second) =
+            if (bytes[0] >= 80) {
+                BigInteger.fromUByte(2u) to BigInteger.fromUInt(bytes[0].toUByte() - 80u)
+            } else {
+                BigInteger.fromUInt(bytes[0].toUByte() / 40u) to BigInteger.fromUInt(bytes[0].toUByte() % 40u)
+            }
+        var index = 1
+        val collected = mutableListOf(first, second)
+        while (index < bytes.size) {
+            if (bytes[index] >= 0) {
+                collected += BigInteger.fromUInt(bytes[index].toUInt())
+                index++
+            } else {
+                val currentNode = mutableListOf<Byte>()
+                while (bytes[index] < 0) {
+                    currentNode += bytes[index] //+= parsed
+                    index++
+                }
+                currentNode += bytes[index]
+                index++
+                collected += currentNode.decodeAsn1VarBigInt().first
+            }
+        }
+        collected
+    }
+
+    /**
+     * Creates an OID in the 2.25 subtree that requires no formal registration.
+     * E.g. the UUID `550e8400-e29b-41d4-a716-446655440000` results in the OID
+     * `2.25.113059749145936325402354257176981405696`
+     */
+    @OptIn(ExperimentalUuidApi::class)
+    constructor(uuid: Uuid) : this(
+        byteArrayOf((2 * 40 + 25).toUByte().toByte(), *uuid.toBigInteger().toAsn1VarInt()), dontVerify = true
+    )
+
+    /**
+     * @param nodes OID Tree nodes passed in order (e.g. 1u, 2u, 96u, …)
+     * @throws Asn1Exception if less than two nodes are supplied, the first node is >2 or the second node is >39
+     */
+    constructor(vararg nodes: UInt) : this(
+        nodes.toOidBytes(),
+        dontVerify = true
+    )
+
+    /**
+     * @param nodes OID Tree nodes passed in order (e.g. 1, 2, 96, …)
+     * @throws Asn1Exception if less than two nodes are supplied, the first node is >2, the second node is >39 or any node is negative
+     */
+    constructor(vararg nodes: BigInteger) : this(bytes = null, nodes = nodes.asList())
```

Comment:
```suggestion
    constructor(vararg nodes: BigInteger) : this(nodes = nodes.asList())
```

Reply:
nope. making this work would require the default constructor parameters to represent an illegal state. not a fan of this strategy

Reply:
did the opposite, now requiring explicitly declaring nulls either way